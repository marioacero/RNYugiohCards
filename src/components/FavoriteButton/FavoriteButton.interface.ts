export interface FavoriteButtonProps {
  isFavorite: boolean;
  onPress: () => void;
}
