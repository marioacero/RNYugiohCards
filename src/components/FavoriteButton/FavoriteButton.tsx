import React, { FC } from 'react';
import { TouchableOpacity } from 'react-native';

import { COLORS } from 'src/utils/colors';
import { FavoriteHeartEmptyIcon, FavoriteHeartFullIcon } from 'src/utils/icons';

import { FavoriteButtonProps } from './FavoriteButton.interface';

const FavoriteButton: FC<FavoriteButtonProps> = ({ isFavorite, onPress }) => {
  const Icon = isFavorite ? FavoriteHeartFullIcon : FavoriteHeartEmptyIcon;
  return (
    <TouchableOpacity onPress={onPress} activeOpacity={1}>
      <Icon width={25} height={25} fill={COLORS.white} />
    </TouchableOpacity>
  );
};

export default FavoriteButton;
