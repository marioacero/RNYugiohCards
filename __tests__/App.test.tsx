import React from 'react';
import FavoriteButton from 'src/components/FavoriteButton';
import { cleanup, render } from '@testing-library/react-native';

describe('FavoriteButton', () => {
  afterEach(cleanup);

  test('should render empty favorite button ', () => {
    const tree = render(
      <FavoriteButton isFavorite={false} onPress={jest.fn} />
    );
    expect(tree).toMatchSnapshot();
  });

  test('should render filled favorite button ', () => {
    const tree = render(<FavoriteButton isFavorite={true} onPress={jest.fn} />);
    expect(tree).toMatchSnapshot();
  });
});
