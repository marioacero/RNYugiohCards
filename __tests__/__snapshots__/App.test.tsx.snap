// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton should render empty favorite button  1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <FavoriteHeartFullIcon
    fill="#ffffff"
    height={25}
    width={25}
  />
</View>
`;

exports[`FavoriteButton should render filled favorite button  1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <FavoriteHeartFullIcon
    fill="#ffffff"
    height={25}
    width={25}
  />
</View>
`;
